From f5109d699511314f0830ea8b279348e26ae119a4 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sun, 20 Aug 2023 13:42:12 +0200
Subject: [PATCH] arm64: dts: imx8mp-hummingboard-pulse: swap m.2 reset and
 w_disable IOs

According to schematics the M.2 reset signal is controlled by
GPIO1_IO06, while W_DISABLE uses GPIO1_IO13.
Swap usage accordingly in rfkill and pinctrl.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../dts/freescale/imx8mp-hummingboard-pulse.dtsi | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dtsi
index b088454dcf10..ba1771f9cde7 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-hummingboard-pulse.dtsi
@@ -106,7 +106,7 @@ rfkill_m2_wdis {
 		pinctrl-0 = <&pinctrl_m2_rfkill>;
 		rfkill-name = "m2_wdis";
 		rfkill-type = <5>;
-		shutdown-gpio = <&gpio1 6 GPIO_ACTIVE_HIGH>;
+		shutdown-gpio = <&gpio1 13 GPIO_ACTIVE_HIGH>;
 		status = "okay";
 	};
 
@@ -378,7 +378,7 @@ dsim_to_adv7535: endpoint {
 &pcie{
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie>;
-	reset-gpio = <&gpio1 13 GPIO_ACTIVE_LOW>;
+	reset-gpio = <&gpio1 6 GPIO_ACTIVE_LOW>;
 	ext_osc = <0>;
 	clocks = <&clk IMX8MP_CLK_HSIO_ROOT>,
 		 <&clk IMX8MP_CLK_PCIE_AUX>,
@@ -546,7 +546,7 @@ MX8MP_IOMUXC_NAND_DATA03__FLEXSPI_A_DATA03	0x82
 
 	pinctrl_m2_rfkill: m2_rfkill_grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO06__GPIO1_IO06		0x19
+			MX8MP_IOMUXC_GPIO1_IO13__GPIO1_IO13		0x19
 		>;
 	};
 
@@ -575,11 +575,11 @@ MX8MP_IOMUXC_GPIO1_IO08__GPIO1_IO08		0x16
 		>;
 	};
 
-        pinctrl_pcie: pciegrp {
-                fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO13__GPIO1_IO13		0x19
-                >;
-        };
+    pinctrl_pcie: pciegrp {
+        fsl,pins = <
+			MX8MP_IOMUXC_GPIO1_IO06__GPIO1_IO06		0x19
+        >;
+    };
 
 	pinctrl_pdm: pdmgrp {
 		fsl,pins = <
-- 
2.35.3
